src/core/unit.c

```diff
@@ -3319,12 +3319,16 @@ int unit_set_default_slice(Unit *u) {
                 if (MANAGER_IS_SYSTEM(u->manager))
                         slice_name = strjoina("system-", escaped, ".slice");
                 else
-                        slice_name = strjoina(escaped, ".slice");
-        } else
-                slice_name =
-                        MANAGER_IS_SYSTEM(u->manager) && !unit_has_name(u, SPECIAL_INIT_SCOPE)
-                        ? SPECIAL_SYSTEM_SLICE
-                        : SPECIAL_ROOT_SLICE;
+                        slice_name = strjoina("app-", escaped, ".slice");
+        } else {
+                if (MANAGER_IS_SYSTEM(u->manager))
+                        slice_name =
+                                !unit_has_name(u, SPECIAL_INIT_SCOPE)
+                                ? SPECIAL_SYSTEM_SLICE
+                                : SPECIAL_ROOT_SLICE;
+                else
+                        slice_name = SPECIAL_APP_SLICE;
```

Comment:
This default would apply to units that don't specify a slice. In my understanding those would be processes either fitting under `session.slice` (such as dbus daemon, keyring damons/agents) or `background.slice` (custom user-defined jobs, monitoring, e-mail sync). What is the reason for the `app.slice`? (Maybe keeping the `-.slice` would be the best at this moment.)

Reply:
The assumption is that we'll have a limited amount of jobs that go into `session.slice`. The rest of what the user starts fits into either `app` or `background`, and if we pick between the two, `app` is a better choice. This is somewhat similar to how we normally start programs from the command-line: in the foreground by default if nothing else is specified.

Reply:
IIUC, you take in from the POV of the launcher (`StartTransientUnit()` DBus call) -- I agree with the foreground semantics. But what about static (user) unit files? Could it go into NEWS that their default location will be changed?

Reply:
> IIUC, you take in from the POV of the launcher (`StartTransientUnit()` DBus call) -- I agree with the foreground semantics.

I was considering any way to start a service (unit file, systemd-run --user, glib wrappers, whatever). *How* it is started is something of an implementation detail.

> But what about static (user) unit files? Could it go into NEWS that their default location will be changed?

Yep, any change like this will need to be described in NEWS. But we now have the practice that NEWS is constructed before the first prerelease. Trying to update NEWS from individual PRs only leads to an endless series of merge conflicts.

Reply:
with the perpetual check in unit_is_extrinsic() (as suggested above) then SPECIAL_INIT_SCOPE will also be considered extrinsic automatically (since it is already perpetual), and the if conditoins will be simpler her.